base/lock.jl

```diff
@@ -51,6 +51,20 @@ end
 
 assert_havelock(l::ReentrantLock) = assert_havelock(l, l.locked_by)
 
+show(io::IO, ::ReentrantLock) = print(io, ReentrantLock, "()")
+
+function show(io::IO, ::MIME"text/plain", l::ReentrantLock)
+    show(io, l)
+    if !(get(io, :compact, false)::Bool)
+        locked_by = l.locked_by
+        if locked_by isa Task
+            print(io, " (locked by ", locked_by === current_task() ? "current " : "", locked_by, ")")
```

Comment:
BTW, `current` doesn't show up in the REPL, even if the code `locked_by === current_task()` executed in the REPL returns true. Maybe the REPL printing is occurring on another task or something?

(Also, the test I updated which includes `current` does pass).